x/concentrated-liquidity/internal/math/math.go

```diff
@@ -12,9 +14,13 @@ func Liquidity0(amount sdk.Int, sqrtPriceA, sqrtPriceB sdk.Dec) sdk.Dec {
 	if sqrtPriceA.GT(sqrtPriceB) {
 		sqrtPriceA, sqrtPriceB = sqrtPriceB, sqrtPriceA
 	}
-	product := sqrtPriceA.Mul(sqrtPriceB)
-	diff := sqrtPriceB.Sub(sqrtPriceA)
-	return amount.ToDec().Mul(product).Quo(diff)
+	amountBigDec := osmomath.BigDecFromSDKDec(amount.ToDec())
```

Comment:
Let's add a comment why we are making these conversions to `BigDec` please. That is, to avoid frequent precision loses for callers of this function